cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -3862,6 +3919,17 @@ TEST(TestArrowReaderAdHoc, CorruptedSchema) {
   TryReadDataFile(path, ::arrow::StatusCode::IOError);
 }
 
+TEST(TestArrowParquet, LargeByteArray) {
+  auto path = test::get_data_file("chunked_string_map.parquet");
+  TryReadDataFile(path, ::arrow::StatusCode::NotImplemented);
+  ArrowReaderProperties reader_properties;
+  reader_properties.set_use_large_binary_variants(true);
+  reader_properties.set_read_dictionary(0, false);
```

Comment:
Honestly I don't think it'll be a problem, file is currently at 4.2KB. I consider it very small, but ofc that's on your side.

I just do not know how to produce such file, the current script that generates the file is the below. It would be great if you could provide an example

```
import pyarrow as pa
import pyarrow.parquet as pq

arr = pa.array([[("a" * 2**30, 1)]], type = pa.map_(pa.string(), pa.int32()))
arr = pa.chunked_array([arr, arr])
tab = pa.table({ "arr": arr })

pq.write_table(tab, "test.parquet", compression='BROTLI')
```

Reply:
Sorry I am not good at python and pyarrow. I did some research and found this doc: https://arrow.apache.org/docs/python/generated/pyarrow.parquet.write_table.html 

Following code snippets may help:

```
pq.write_table(table, 'example.parquet',
               compression={'n_legs': 'snappy', 'animal': 'gzip'},
               use_dictionary=['n_legs', 'animal'])

pq.write_table(table, 'example.parquet',
               column_encoding={'animal':'PLAIN'},
               use_dictionary=False)
```

Reply:
Wouldn't this produce two different files instead of a single file with a single table and two columns?

Reply:
```
import pyarrow as pa
import pyarrow.parquet as pq

arr = pa.array([[("a" * 2**30, 1)]], type = pa.map_(pa.string(), pa.int32()))
arr = pa.chunked_array([arr, arr])
tab = pa.table({ "arr": arr, "dict_arr": arr})

pq.write_table(tab, "test.parquet", compression='BROTLI', use_dictionary=['dict_arr'])
```

I have written the above script, but I am not sure it's doing what it's supposed to do. Is there a way to verify it? `parquet-tools inspect` does not tell me much.

@mapleFU @wgtmac

Reply:
You may need to build https://github.com/apache/arrow/blob/main/cpp/tools/parquet/parquet_reader.cc to print the metadata. It will tell you whether the column is dictionary encoded or not. Please make sure `PARQUET_BUILD_EXECUTABLES` is set to `ON` as it is disabled by default.

Reply:
Does not look like encoding was added:

```
arthur@arthur:~/parquet-testing$ ~/arrow/cpp/cmake-build-ninja-debug/debug/parquet-reader ~/parquet-testing/test.parquet --only-metadata --print-key-value-metadata
File Name: /home/arthur/parquet-testing/test.parquet
Version: 2.6
Created By: parquet-cpp-arrow version 11.0.0
Total rows: 2
Key Value File Metadata: 1 entries
 Key nr 0 ARROW:schema: /////8gBAAAQAAAAAAAKAAwABgAFAAgACgAAAAABBAAMAAAACAAIAAAABAAIAAAABAAAAAIAAAC8AAAABAAAAPT+//8AAAERFAAAACAAAAAEAAAAAQAAABgAAAAIAAAAZGljdF9hcnIAAAAAqP7//9z+//8AAAANGAAAACAAAAAEAAAAAgAAAEgAAAAUAAAABwAAAGVudHJpZXMA2P7//1T///8AAAECEAAAABgAAAAEAAAAAAAAAAUAAAB2YWx1ZQAAAET///8AAAABIAAAADz///8AAAAFEAAAABQAAAAEAAAAAAAAAAMAAABrZXkALP///6j///8AAAERFAAAABgAAAAEAAAAAQAAABAAAAADAAAAYXJyAFT///+I////AAAADRgAAAAgAAAABAAAAAIAAABwAAAAJAAAAAcAAABlbnRyaWVzAIT///8QABQACAAGAAcADAAAABAAEAAAAAAAAQIQAAAAIAAAAAQAAAAAAAAABQAAAHZhbHVlAAAACAAMAAgABwAIAAAAAAAAASAAAAAQABQACAAAAAcADAAAABAAEAAAAAAAAAUQAAAAGAAAAAQAAAAAAAAAAwAAAGtleQAEAAQABAAAAAAAAAA=
Number of RowGroups: 1
Number of Real Columns: 2
Number of Columns: 4
Number of Selected Columns: 4
Column 0: arr.key_value.key (BYTE_ARRAY / String / UTF8)
Column 1: arr.key_value.value (INT32)
Column 2: dict_arr.key_value.key (BYTE_ARRAY / String / UTF8)
Column 3: dict_arr.key_value.value (INT32)
--- Row Group: 0 ---
--- Total Bytes: 4294967594 ---
--- Total Compressed Bytes: 6774 ---
--- Rows: 2 ---
Column 0
  Values: 2, Null Values: 0, Distinct Values: 0
  Max: , Min: 
  Compression: BROTLI, Encodings: PLAIN
  Uncompressed Size: 2147483732, Compressed Size: 3326
Column 1
  Values: 2, Null Values: 0, Distinct Values: 0
  Max: 1, Min: 1
  Compression: BROTLI, Encodings: PLAIN
  Uncompressed Size: 65, Compressed Size: 61
Column 2
  Values: 2, Null Values: 0, Distinct Values: 0
  Max: , Min: 
  Compression: BROTLI, Encodings: PLAIN
  Uncompressed Size: 2147483732, Compressed Size: 3326
Column 3
  Values: 2, Null Values: 0, Distinct Values: 0
  Max: 1, Min: 1
  Compression: BROTLI, Encodings: PLAIN
  Uncompressed Size: 65, Compressed Size: 61

```

Reply:
Ok.. so that file is ready to be merged, right?

Reply:
It's worth noting that if I simply put `use_dictionary=True`, both columns appear with: ` Compression: BROTLI, Encodings: PLAIN(DICT_PAGE) PLAIN RLE_DICTIONARY
`

Reply:
You probably need to pass a per-column setting?

Reply:
Yes, that's what I tried to do with `pq.write_table(tab, "test.parquet", compression='BROTLI', use_dictionary=['dict_arr'])`. According to the docs, the `use_dictionary` argument can be a list.

Reply:
Aha, thanks for confirming that. Ok, can that file be merged as it is then?